x-pack/legacy/plugins/reporting/server/lib/create_worker.ts

```diff
@@ -33,30 +35,36 @@ export function createWorkerFactory(server: ServerFacade) {
   // Once more document types are added, this will need to be passed in
   return function createWorker(queue: ESQueueInstance) {
     // export type / execute job map
-    const jobExecutors: Map<string, ESQueueWorkerExecuteFn | ImmediateExecuteFn> = new Map();
+    const jobExecutors: Map<string, ExecuteFnType> = new Map();
 
-    for (const exportType of exportTypesRegistry.getAll() as ExportTypeDefinition[]) {
+    for (const exportType of exportTypesRegistry.getAll() as Array<
+      ExportTypeDefinition<JobParamsType, any, any, ExecuteFnType>
+    >) {
       const executeJobFactory = exportType.executeJobFactory(server);
       jobExecutors.set(exportType.jobType, executeJobFactory);
     }
 
     const workerFn = (
-      job: JobSource,
-      arg1: JobDocPayload | JobDoc,
+      { _id: jobId, _source: { jobtype: jobType } }: JobSourceType,
+      arg1: JobDocPayload<JobParamsType> | JobParamsType,
```

Comment:
This signature is plain bonkers. I don't really like the `arg1` and `arg2`. Again, refactor for another time.

Reply:
I know - I don't think there are any bugs in here, for now... it seems like asking for trouble.

Reply:
I think I can make a good number of improvements here before merge

Reply:
Since we only care about arg1 and arg2 to pass along to the "real" execute function, we can use the spread operator for the function signature

Reply:
Can you take another look at 00870d6f1d ?